src/usbasp.c

```diff
@@ -399,6 +399,37 @@ static int usbasp_transmit(const PROGRAMMER *pgm,
   return nbytes;
 }
 
+static int check_for_port_argument_match(const char *port, char *bus, char *device, char *serial_num) {
```

Comment:
This function returns a 1 if the port arguments are *not* matched and 0 if they do. So I suggest renaming the function to `port_argument_not_matched()`.